opaevaluator.go

```diff
@@ -254,16 +256,26 @@ func (evaluator *OPAEvaluator) partiallyEvaluate(logger *logrus.Entry) (primitiv
 	if err != nil {
 		return nil, err
 	}
+	m, err := metrics.GetFromContext(evaluator.Context)
```

Comment:
why collecting metrics if they are not exposed?

I'd set a fake metrics object in the context (done by the middleware if the `env.ExposeMetrics` flag is `false`) so that registering data is a noop and does not accumulate unused data in memory; what do you think?

Reply:
prometheus client does not expose a fake client, so I don't like to create my own fake object. If the object is not exposed, there is a little memory consuming to store the data, not a so big impact.

In case the metrics should not be taken, we can return here an error and avoid to take the information.

Reply:
we could define an interface that wraps used methods and interchange it so that memory is unaffected; rond is not that memory-greedy however I'm worried about useless garbage collection that may have impact on the overall performances.

Reply:
It's the same. Create a no-op implementation can break something if, for example, we will use some not changed method which will break if not used accordingly with previous method.